mureader.py

```diff
@@ -0,0 +1,679 @@
+            self.mainTex = MuMatTex().read(mu)
+            self.bumpMap = MuMatTex().read(mu)
+            self.specColor = mu.read_float(4)
+            self.shininess = mu.read_float()
+            self.emissive = MuMatTex().read(mu)
+            self.emissiveColor = mu.read_float(4)
+        elif self.type == MuEnum.ST_ALPHA_CUTOUT:
+            self.mainTex = MuMatTex().read(mu)
+            self.cutoff = mu.read_float()
+        elif self.type == MuEnum.ST_ALPHA_CUTOUT_BUMPED:
+            self.mainTex = MuMatTex().read(mu)
+            self.bumpMap = MuMatTex().read(mu)
+            self.cutoff = mu.read_float()
+        elif self.type == MuEnum.ST_ALPHA:
+            self.mainTex = MuMatTex().read(mu)
+        elif self.type == MuEnum.ST_ALPHA_SPECULAR:
+            self.mainTex = MuMatTex().read(mu)
+            self.gloss = mu.read_float()
+            self.specColor = mu.read_float(4)
+            self.shininess = mu.read_float()
+        elif self.type == MuEnum.ST_ALPHA_UNLIT:
+            self.mainTex = MuMatTex().read(mu)
+            self.color = mu.read_float(4)
+        elif self.type == MuEnum.ST_UNLIT:
+            self.mainTex = MuMatTex().read(mu)
+            self.color = mu.read_float(4)
+        elif self.type == MuEnum.ST_DIFFUSE:
+            self.mainTex = MuMatTex().read(mu)
+        else:
+            raise ValueError("MuMaterial %d" % self.type)
+        return self
+
+
+class MuTransform:
+    def __init__(self):
+        pass
+    def read(self, mu):
+        self.name = mu.read_string()
+        self.localPosition = mu.read_vector()
+        self.localRotation = mu.read_quaternion()
+        self.localScale = mu.read_vector()
+        #      self.localScale)
+        return self
+
+
+class MuTagLayer:
+    def __init__(self):
+        pass
+    def read(self, mu):
+        self.tag = mu.read_string()
+        self.layer = mu.read_int()
+        return self
+
+
+class MuKey:
+    def __init__(self):
+        pass
+    def read(self, mu):
+        self.time = mu.read_float()
+        self.value = mu.read_float()
+        self.tangent = mu.read_float(2) # in, out
+        self.tangentMode = mu.read_int()
+        # editable, smooth, linear, stepped (0..3?)
+        return self
+
+
+class MuCurve:
+    def __init__(self):
+        pass
+    def read(self, mu):
+        self.path = mu.read_string()
+        self.property = mu.read_string()
+        self.type = mu.read_int()
+        self.wrapMode = mu.read_int(2)  # pre, post
+        num_keys = mu.read_int()
+        self.keys = []
+        for i in range(num_keys):
+            self.keys.append(MuKey().read(mu))
+        return self
+
+
+class MuClip:
+    def __init__(self):
+        self.curves = []
+    def read(self, mu):
+        self.name = mu.read_string()
+        self.lbCenter = mu.read_vector()
+        self.lbSize = mu.read_vector()
+        self.wrapMode = mu.read_int()
+        num_curves = mu.read_int()
+        for i in range(num_curves):
```

Comment:
Use a list comprehension with silent variable
`self.curves = [MuCurve().read(mu) for _ in range(num_curves)]` (or with my previous remark `self.curves = [MuCurve(mu) for _ in range(num_curves)]`